Fix `RunFilterParameter` doesn't use the default value

ikedam opened the pull request:
Add tests for `RunFilterParameter` and `ParameterizedRunFilter`, and it discovered a bug in `RunFilterParameter`.
I forgot to override `SimpleParameterDefinition#getDefaultParameterValue` and it caused default values of `RunFilterParameter` not used at all.

ikedam committed:
Add tests for RunFilterParameter and ParameterizedRunFilter.

ikedam commented on pom.xml:
```diff
@@ -43,6 +43,13 @@
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins.workflow</groupId>
```
For `StepConfigTester` which allows me test that `DataBoundConstructor` works as expected.

ikedam commented:
https://jenkins.ci.cloudbees.com/job/plugins/job/run-selector-plugin/org.jenkins-ci.plugins$run-selector/7/testReport/junit/org.jenkinsci.plugins.runselector.filters/ParameterizedRunFilterTest/testIsSelectableWithDefault/
Test failed for the bug.
The default value isn't used at all.
I'll add a commit to fix it.

ikedam committed:
Fix RunFilterParameter doesn't use default values.

alexsomai commented on src/test/java/org/jenkinsci/plugins/runselector/filters/ParameterizedRunFilterTest.java:
```diff
@@ -0,0 +1,252 @@
+    @ClassRule
+    public static final JenkinsRule j = new JenkinsRule();
+
+    private static WorkflowJob jobToCopyFrom;
```
Since we are not in the Copy Artifact plugin context, I would rename this maybe to `jobToSelect`, or `upstreamJob`. Same for the `runToCopyFrom`. But this is not very important.

ikedam replied:
You are absolutely right!
I used the word "copy" from habit... I fixed them.

alexsomai commented on src/test/java/org/jenkinsci/plugins/runselector/filters/ParameterizedRunFilterTest.java:
```diff
@@ -0,0 +1,252 @@
+            RandomStringUtils.randomAlphanumeric(7)
+        );
+        selecter.addProperty(new ParametersDefinitionProperty(
+            Arrays.<ParameterDefinition>asList(new RunFilterParameter(
```
Usually when I'm creating a list with one element, I prefer using `Collections#singletonList`, rather than `Arrays#asList`. But I don't have any strong opinion for this.

ikedam replied:
I tested `Collections.singletonList` and found that `ParameterDefinitionsProperty` provides a constractor with a variant number of parameters!
I don't need `Array.asList` nor `Collection.singletonList`.
(I used unnecessary `Array.asList` everytime I use `ParameterDefinitionsProperty` for long time...)
Thank you for let me find that.

alexsomai commented:
Thank you for this PR @ikedam 
It looks good to me 👍 
@oleg-nenashev can you please have a quick look too?

ikedam committed:
Improve ParameterizedRunFilterTest.

Use more appropriate words and change "copy" to "select"
Remove Arrays.asList and use constructors with variant arguments instead.
